alexandria/core/serializers.py

```diff
@@ -147,39 +153,100 @@ class FileSerializer(BaseSerializer):
         "renderings": "alexandria.core.serializers.FileSerializer",
     }
 
+    download_url = serializers.SerializerMethodField()
+
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         # We only want to provide an upload_url on creation of a file.
         self.new = False
         if "request" in self.context and self.context["request"].method == "POST":
             self.new = True
 
-    download_url = serializers.CharField(read_only=True)
-    upload_url = serializers.SerializerMethodField()
-
-    def get_upload_url(self, obj):
-        return obj.upload_url if self.new else ""
+    @staticmethod
+    def make_signature_components(
+        pk: str, request: Request, expires: Optional[int] = None
+    ) -> Tuple[str, int, str]:
```

Comment:
I think this method could be moved into a module context, so it's accessible without the DRF componentry

Reply:
If I move it to its own module would you also suggest that I make it non-dependent on the `Request` object? Requiring a host parameter? That was actually the reason to add it there, because that's a context where the request is available normally

Reply:
Yes I think that'd be a good idea. Might make the code a bit cleaner too

Reply:
done  in fdb5943